Fork_Maintenance.md

```diff
@@ -0,0 +1,263 @@
+
+```bash
+git fetch upstream
+```
+
+**b. Review Upstream Changes:**
+
+Before integrating, review the changes made upstream since your `main` branch was last updated.
+
+```bash
+# Ensure you are on your local main branch
+git checkout main
+
+# See commit history difference
+git log HEAD..upstream/main --oneline --graph
+
+# See file changes
+git diff HEAD..upstream/main
+```
+Pay close attention to changes in core components, interfaces between `agent` and `runtime`, dependencies, and security-related areas.
+
+**c. Update Local `main` Branch (Rebase Recommended):**
+
+Rebase your local `main` branch onto `upstream/main` to maintain a clean, linear history mirroring the upstream repository.
+
+```bash
+git checkout main
+git rebase upstream/main
+```
+
+**Conflict Resolution (Main):** Conflicts here should be rare if you are not making direct commits to `main`. If they occur, resolve them, `git add <file>`, and `git rebase --continue`.
+
+**Push Updated `main`:** Push the updated `main` branch to your fork's origin. Use `--force-with-lease` because you rebased.
+
+```bash
+git push origin main --force-with-lease
+```
+
+**d. Integrate Upstream Changes into `release/stable-with-patches` (Rebase):**
+
+Regularly rebase your `release/stable-with-patches` branch onto the updated `main` branch. This incorporates the latest upstream changes while keeping your custom patches on top.
+
+```bash
+git checkout release/stable-with-patches
+git rebase main
+```
+
+**Conflict Resolution (Release Branch):** Conflicts are more likely here, as upstream changes might clash with your custom patches. Carefully resolve conflicts, ensuring both upstream updates and your custom logic are correctly merged. Use `git status` to see conflicted files, edit them, then `git add <file>` and `git rebase --continue`. If you get stuck, `git rebase --abort` cancels the rebase.
+
+**e. Testing:**
+
+After rebasing `release/stable-with-patches`, thoroughly test your fork. Run linters, unit tests, integration tests, and perform manual checks focusing on areas affected by both upstream changes and your custom patches.
+
+```bash
+# Example: Run pre-commit checks and tests (adapt as needed)
+make install-pre-commit-hooks
+pre-commit run --all-files --config ./dev_config/python/.pre-commit-config.yaml
+# cd frontend && npm run lint:fix && npm run build ; cd ..
+# poetry run pytest ...
+```
+
+**f. Push Updated `release/stable-with-patches`:**
+
+Push the rebased `release/stable-with-patches` branch to your fork's origin. Use `--force-with-lease` because you rebased.
+
+```bash
+git push origin release/stable-with-patches --force-with-lease
+```
+
+**This push will typically trigger your CI/CD pipeline to build and deploy new container images (as described in Section 2).**
+
+## 4. Custom Development Workflow
+
+All custom development and integration of external patches happen relative to the `release/stable-with-patches` branch.
+
+**a. Create Feature Branches:**
```

Comment:
Honestly, workflow-related changes are a pain to test. We might want another test app `acme-openhands` that always tracks `release/experimental` or something that we can openly push to in order to quickly test flows.